Allow canceling in progress orchestrations (#355)
kyma-project/control-plane: docs/kyma-environment-broker/03-10-orchestration.md

Orchestration is a mechanism that allows you to upgrade Kyma Runtimes. To create an orchestration, [follow this tutorial](#tutorials-orchestrate-kyma-upgrade). After sending the request, the orchestration is processed by `KymaUpgradeManager`. It lists Shoots (Kyma Runtimes) in the Gardener cluster and narrows them to the IDs that you have specified in the request body. Then, `KymaUpgradeManager` performs the [upgrade steps](#details-runtime-operations) logic on the selected Runtimes.

> If Kyma Environment Broker is restarted, it reprocesses the orchestration with the `CANCELING`, `IN PROGRESS`, and `PENDING` state.

Reviewer: ```suggestion
If Kyma Environment Broker is restarted, it reprocesses the orchestrations that were previously in the `CANCELING`, `IN PROGRESS`, and `PENDING` state.
```

Orchestration API consist of the following handlers:

- `GET /orchestrations` - exposes data about all orchestrations.

> - `GET /orchestrations/{orchestration_id}` - exposes status of a single orchestration.

Reviewer: ```suggestion
- `GET /orchestrations/{orchestration_id}` - exposes the status of a single orchestration.
```

  }
}
```

## Cancelation

> You can cancel any orchestration that is in progress or pending using the `PUT /orchestrations/{orchestration_id}/canceled` endpoint described above.

Reviewer: ```suggestion
You can cancel any orchestration that is in progress or pending using the `PUT /orchestrations/{orchestration_id}/canceled` endpoint.
```

After you cancel an orchestration, the KEB sets its state to `Canceled`. Orchestration with such state is not scheduling any new operations. Every pending operation in case of cancel are also marked with `Canceled` state.

> To provide consistency, cancelled orchestration waits for already processed operations to finish. When operations are finished the orchestration queue processing becomes unblocked.

Reviewer: ```suggestion
To provide consistency, a canceled orchestration waits for already processed operations to finish. When operations are finished, the cancelation is processed.
```

  }
}
```

## Cancelation

You can cancel any orchestration that is in progress or pending using the `PUT /orchestrations/{orchestration_id}/cancel` endpoint. 
After you cancel an orchestration, KEB sets its state to `Canceling`. An orchestration with such a state does not schedule any new operations.

> To provide consistency, a canceled orchestration waits for already processed operations to finish. When operations are finished, the processed orchestration's state is set to `Canceled` and the next orchestration from the queue is processed.

Reviewer: ```suggestion
To provide consistency, a canceled orchestration waits for already processed operations to finish. When operations are finished, the processed orchestration's state is set to `Canceled` and the next orchestration from the queue starts being processed.
```

> To provide consistency, a canceled orchestration waits for already processed operations to finish. When operations are finished, the processed orchestration's state is set to `Canceled` and the next orchestration from the queue is start to being processed.

Reviewer: ```suggestion
To provide consistency, a canceled orchestration waits for already processed operations to finish. When operations are finished, the processed orchestration's state is set to `Canceled` and the next orchestration from the queue starts to be processed.
```
